Source/Core/VideoBackends/OGL/VertexManager.cpp

```diff
@@ -140,6 +140,15 @@ void VertexManager::vFlush(bool useDstAlpha)
 
 	PrepareDrawBuffers(stride);
 
+	if (!bpmem.genMode.zfreeze && IndexGenerator::GetIndexLen() >= 3)
+	{
+		CalculateZSlope(stride);
```

Comment:
I'm not sure.
it's possibly cheaper to just calculate the zslope and store the slope. (If you are copying the matrix, doing the transformation is not much more expensive and you save write memory brandwidth.)

I suspect what is currently expensive is the excessive updating of uniform constants every single drawcall, we should only update the zslope constants when zfreeze is actually enabled.